service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionController.java

```diff
@@ -1,34 +1,41 @@
 package uk.gov.hmcts.reform.fpl.controllers;
 
 import io.swagger.annotations.Api;
+import org.json.JSONException;
+import org.json.JSONObject;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
-import uk.gov.hmcts.reform.fpl.templates.DocumentTemplates;
-import uk.gov.hmcts.reform.pdf.generator.HTMLToPDFConverter;
-
-import java.util.Map;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.fpl.service.CaseService;
 import javax.validation.constraints.NotNull;
+import java.io.IOException;
+import java.util.Map;
 
 @Api
 @RestController
 @RequestMapping("/callback/case-submission")
 public class CaseSubmissionController {
 
-    private final HTMLToPDFConverter converter = new HTMLToPDFConverter();
-
     @Autowired
-    private DocumentTemplates documentTemplates;
+    private CaseService caseService;
 
     @PostMapping
-    public ResponseEntity submittedCase(@RequestBody @NotNull Map<String,
-        Object> caseData) {
-        byte[] template = documentTemplates.getHtmlTemplate();
-        byte[] pdfDocument = converter.convert(template, caseData);
+    public ResponseEntity submittedCase(
+        @RequestHeader(value = "authorization") String authorization,
+        @RequestHeader(value = "user-id") String userId,
+        @RequestBody @NotNull CallbackRequest caseData) throws JSONException, IOException {
+        System.out.println("Authorization: " + authorization);
+        System.out.println("User Id: " + userId);
+        System.out.println("Case data: " + caseData);
+
+        caseService.handleCaseSubmission(authorization, userId, caseData);
```

Comment:
Please use spring events to notify consumers that case got submitted. See https://www.baeldung.com/spring-events for reference.

In nutshell:
- controller publishes event `CaseSubmitted` with every available details
- implemented consumer handles PDF generation, document upload and case update (your story)
- when we introduce email notifications another consumer will handle sending emails (notifications story - future)

Reply:
On a second though provided blog link suggests to use interfaces while newer approach relies on annotation which I prefer. Here is sample consumer with annotation:


```
@Component
public class CitizenActionsHandler {
    
    @Autowired
    private NotificationService notificationService;

    @EventListener
    public void sendNotification(CountyCourtJudgmentEvent event) {
        Claim claim = event.getClaim();
        ....
```

☝️`@EventListener` simply makes class a consumer.